acceptance/indy_acceptance/test_scripts/test_scenario_base.py

```diff
@@ -0,0 +1,102 @@
+    test scenarios that inherit it.
+    """
+    pool_name = generate_random_string("test_pool")
+    wallet_name = generate_random_string("test_wallet")
+    pool_handle = 0
+    wallet_handle = 0
+    pool_genesis_txn_file = constant.pool_genesis_txn_file
+    logger = None
+    steps = None
+    test_result = None
+    test_name = ""
+
+    def init_data_test(self):
+        """
+        Init test data.
+        If the test case need some extra test date
+        then just override this method.
+        """
+        self.test_name = os.path.splitext(
+            os.path.basename(inspect.getfile(self.__class__)))[0]
+        self.test_result = TestResult(self.test_name)
+        self.steps = Steps()
+        self.logger = Logger(self.test_name)
+
+    async def execute_precondition_steps(self):
+        """
+         Execute pre-condition of test scenario.
+         If the test case need some extra step in pre-condition
+         then just override this method.
+        """
+        common.clean_up_pool_and_wallet_folder(
+            self.pool_name, self.wallet_name)
+
+    async def execute_postcondition_steps(self):
+        """
+        Execute post-condition of test scenario.
+        If the test case need some extra step in post-condition
+        then just override this method.
+        """
+        await common.clean_up_pool_and_wallet(self.pool_name, self.pool_handle,
+                                              self.wallet_name,
+                                              self.wallet_handle)
+
+    async def execute_test_steps(self):
+        """
+        The method where contain all main script of a test scenario.
+        All test scenario inherit TestScenarioBase have to override this method
+        """
+        pass
+
+    def execute_scenario(self):
+        """
+        Execute the test scenario and control the work flow of
+        this test scenario.
+        """
+        begin_time = time.time()
+        self.init_data_test()
+        utils.print_with_color("\nTest case: {} ----> started\n"
+                               .format(self.test_name), Color.BOLD)
+        try:
+            run_async_method(self.execute_precondition_steps)
+            run_async_method(self.execute_test_steps)
+        except Exception as e:
+            message = constant.EXCEPTION.format(str(e))
+            utils.print_error("\n{}\n".format(str(message)))
+            self.steps.get_last_step().set_status(Status.FAILED, message)
+        finally:
+            try:
+                run_async_method(self.execute_postcondition_steps)
+            except Exception as e:
+                utils.print_error("\n{}\n".format(str(type(e))))
+            make_final_result(self.test_result, self.steps.get_list_step(),
+                              begin_time, self.logger)
+            utils.print_with_color("Test case: {} ----> finished\n".
+                                   format(self.test_name), Color.BOLD)
```

Comment:
Now "finished" message may be misinterpreted because it is printed for any test result: success or fail.
It would be better not to print "finished" but just to print "passed" in case of success and "failed" with the occurred error in case of fail.

Reply:
Will update.
FYI, we printed the error message in red on the console for the failure and there will be a failed record in the log file.